packages/domainobjs/ts/voteCounts.ts

```diff
@@ -0,0 +1,127 @@
+import { hash5, hashLeftRight, IncrementalQuinTree } from "@maci-protocol/crypto";
+
+import assert from "assert";
+
+import type { IJsonVoteCounts } from "./types";
+
+/**
+ * A VoteCounts represents a User's vote counts in a Poll, as well as their next valid
```

Comment:
why is it for each user? what if it was for all users in one structure (i.e. just keeping the most recent count with the index being the option and the value how many voters chose the option)?

Reply:
It's the same as for ballot. The only one difference that we need to calculate all the vote counts and build a tree (similar to ballot tree). So that's why nonce is required.